### DISCUSSION TITLE
Who is responsible for removing the cni config file and cni binary of istio-cni

### DISCUSSION THREAD

**Discussion by wyike** [istio/istio#50022] [Q&A]
install-cni containers of istio-cni-node pod will install cni config file and binary.
Who is responsible to remove them? And what's the logical of the removal?

Thanks!

**Comment by howardjohn** [comment]
istio-cni-node removes them on shutdown. This will, of course, not work on a forceful shutdown

**Comment by wyike** [comment]
Thanks John! 
Yeah, when I delete the istio-cni-node, the files are just deleted. How this mechanism can happen? Could you explain more and show the code if possible? Thanks a lot.

Can the removal have a chance to lead to a situation: the conflist is still there while the binary is deleted?
Recently I met a customer issue, their env only has confilist but no binary. I only find https://github.com/istio/istio/pull/36831 this can be lead to such a situation but during creation.

**Comment by howardjohn** [comment]
I believe it's watching SIGTERM and cleans up

**Reply by wyike** [reply]
Thanks John. From my test, it looks so.

In my impression, the traditional cni agent always lets init-container to install cni config/bin and also runs the pod in host Network mode. This best practice should avoid the case that a cni conf list configured while cni bin is missing (for any reason, folder is changed, overwrite or related disk detach, any bug like https://github.com/istio/istio/pull/36831 etc.) and hence all the pod on the system cannot start include istio-cni-node itself. 
So I am wondering that is there any specific reason that istio-cni removes hostNetwork support https://discuss.istio.io/t/istio-cni-for-1-11-4-doesnt-need-hostnetwork-true-anymore/11603? Thanks!

**Comment by wyike** [comment]
hi @howardjohn  would you help point the istio node cleans up code with SIGTERM?  Thanks a million!

**Reply by howardjohn** [reply]
https://github.com/istio/istio/blob/fb9179422d298dce8fcb50064fbe3132dea57176/cni/pkg/cmd/root.go#L131
